### DISCUSSION TITLE
Install GitHub dependency that uses requirements.txt

### DISCUSSION THREAD

**Discussion by mywristbands** [python-poetry/poetry#7199] [Q&A]
I'd like to install [whisper](https://github.com/openai/whisper) using poetry. I tried doing this by adding the following line to my `pyproject.toml` file:
```
whisper = {git = "https://github.com/openai/whisper.git", rev = "0b5dcfdef7ec04250b76e13f1630e32b0935ce76"}
```

And then I ran `poetry update whisper`.

While this does work to install whisper, it doesn't install any of whisper's dependencies listed in its `requirements.txt`. I _could_ manually download all of these dependencies, but this is really something I would expect my package manager to do. Is it possible to use poetry to automatically download whisper _and_ its dependencies?

Or, would I need to switch to another python package dependency manager (such as pipenv) in order to easily install this package?

**Comment by neersighted** [comment]
Poetry will use the metadata reported by the package's build-system, in this case setuptools. It appears that whisper is attempting to expose their requirements.txt in their metadata (https://github.com/openai/whisper/blob/main/setup.py#L17-L22); building wheels and investigating what metadata is captured in the `METADATA` file would be your next step to debug.

**Reply by mywristbands** [reply]
Thank you for the response! I'm glad it seems you believe this repository should work with poetry.

However, I'm not sure how to debug:
- What do you mean by that I should be "building wheels"? Do you mean that I should build the wheel for whisper by following [instructions like this](https://www.activestate.com/resources/quick-reads/python-install-wheel/)?
- How do I investigate the metadata captured in the `METADATA` file? I don't know what this `METADATA` file is or where it is located.

I did find an unofficial distribution of whisper on PyPI called [whisper.ai](https://pypi.org/project/whisper.ai/). I tried installing it with poetry, and it did install all the dependencies besides PyTorch. So I just manually installed PyTorch, and it's now working. That said, ideally I would like to use the official distribution.

**Comment by mywristbands** [comment]
For anyone else reading this tread, ultimately I decided to manage my python dependencies with pip, pyenv, pyenv-virtualenv, and a `requirements.txt` file. Here's a [post of mine describing how I set it up](https://github.com/openai/whisper/discussions/683#discussioncomment-4587894).

The reason I decided against Poetry was I was having a lot of trouble getting different wheels of the PyTorch library installed in my different environments. These environments included:
- macOS installed on M1 chip
- linux docker container running on my macOS installed on M1 chip
- linux docker container running on an AWS G4 instance

Using pip made it easy to install the correct PyTorch wheel for each environment.

**Reply by tombohub** [reply]
So you never found a solution for this?
